site/src/pages/RoadmapPage/sidebar/SingleMajorCourseList.tsx

```diff
@@ -79,7 +79,10 @@ const SingleMajorCourseList: FC<SingleMajorCourseListProps> = ({
   );
 
   const loadSpecRequirements = useCallback(async () => {
-    if (!hasSpecs) return await fetchRequirements(major.id, null);
+    if (!hasSpecs) {
+      if (majorWithSpec.requirements.length > 0) return;
+      else return await fetchRequirements(major.id, null);
+    }
```

Comment:
We're getting a lint warning with the deps array; see if you can fix the lint warning and still achieve the behavior you're mentioning. If so, we are good to merge the PR I think.

Reply:
Update: made the commit, double check that it works